### ISSUE TITLE
git_odb_write_pack should not require the ODB to understand pack files

### ISSUE THREAD

**Issue by medranocalvo** [libgit2/libgit2#4494]
The documentation for `git_odb_write_pack` claims that:

> If the ODB layer does not understand pack files, the objects will be stored in whatever format the ODB layer uses.

That is not currently the case: if no ODB backend implements the `writepack` method, the following error comes up upon e.g. `fetch`:

~~~
cannot write pack - unsupported in the loaded odb backends (code 9)
~~~

Although a generic way of implementing support for packfiles in ODBs is described in ticket #3098, having `git_odb_write_pack` dissasemble the packfiles for the ODBs (perhaps using the method above) would reduce burden and errors when implementing custom ODB backends.

Note that default backends `odb_loose` and `odb_mempack` don't support `writepack`.

### Expected behavior

`git_odb_write_pack` disassembles the packfiles into objects and passes those to the ODB backends when no ODB backend supports packfiles.

### Actual behavior

Abort with error `cannot write pack - unsupported in the loaded odb backends (code 9)`.

### Version of libgit2 (release number or SHA1)
0.26.0
### Operating system(s) tested

**Comment by ethomson** [comment]
I agree that this would be a nice to have, at least to unpack for the loose backend.

The `odb_mempack` backend, however, is _not_ a standard backend, it's experimental.  If we can get generic pack -> unpack support, great.  If not, I don't think that having `odb_mempack` support is particularly important.

**Comment by medranocalvo** [comment]
Thank you for your comments.  In my opinion the most helpful behaviour would be for `git_odb_write_pack` to use the functionality the backend happens to provide, similarly to what happens with streams.
